Customer.java

```diff
@@ -3,59 +3,73 @@
 
 import Client,Cinema,Movie;
 
-public class Customer implements Client {
-    private Ticket[] bookings;  //previous bookings made by customer given by tickets.
-    private boolean auth = false;  //authenticatiuon indicator. Whether customer is authenticated or not. All methods must check for authentication.
+public class Customer implements Client, Serializable{
+
+    private int age;
+    private String name;
+    private boolean auth = false;  //authentication indicator. Whether customer is authenticated or not. All methods must check for authentication.
     public String username;
+    public String password;
 
-    public boolean login(String password, Database db) {
-        if (!db.user_accounts.containsKey(this.username)) return false;
+    //Constructor for Customer object.
+    public Customer(String name, int age) {
+        this.name = name;
+        this.age = age;
+    }
 
-        else if (db.user_accounts.get(this.username) != password) return false;
-        this.auth = true;
-        System.out.println("Authenticated succesfully");
-        bookings = new Ticket[100];
-        return true;
+    public Customer(String name, int age, String username, String password) {
+        this.name = name;
+        this.age = age;
+        this.username = username;
+        this.password = password;
     }
 
-    public boolean createAccount(Database db) throws Exception {
+
+    public boolean login(CustomerController c) throws IOException {
+        int tries = 9;
+        String tempUsername;
+        String tempPassword;
         BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));
+
+        System.out.println("Please enter your login username: ");
+        tempUsername = reader.readLine();
+        System.out.println("Please enter your password: ");
+        tempPassword = reader.readLine();
+
+        ArrayList<Customer> customers = null;
```

Comment:
You should use a proper initializer for ArrayList like so

`ArrayList<Customer> customers = new ArrayList<>();`